operator/apis/platform/v1alpha1/central_types.go

```diff
@@ -194,15 +194,12 @@ type CentralDBSpec struct {
 }
 
 // CentralDBEnabled is a type for values of spec.central.db.enabled.
-//+kubebuilder:validation:Enum=Default;Enabled
+//+kubebuilder:validation:Enum=Default
 type CentralDBEnabled string
 
 const (
-	// CentralDBEnabledDefault configures the central to use rocksdb
-	// TODO(ROX-13557): in 3.74.0 this should say that Default enables PostgreSQL.
+	// CentralDBEnabledDefault configures the central to use PostgreSQL database (Technology Preview)
 	CentralDBEnabledDefault CentralDBEnabled = "Default"
-	// CentralDBEnabledTrue configures the central to use a PostgreSQL database (Technology Preview)
-	CentralDBEnabledTrue CentralDBEnabled = "Enabled"
```

Comment:
I also don't think it's a good idea to remove the "Enabled" value. All our existing ACS instances in ACSCS use "Enabled" atm.

I think we should keep both values for now, the only difference being that "Default" now also means "Enabled". After the 3.74 release, we can omit setting this field in fleetshard-sync. Then in the 3.75 release we can remove "isEnabled" altogether.

Reply:
(Disclaimer: I scrolled through comments and randomly decided to respond.)

> Then in the 3.75 release we can remove "isEnabled" altogether.

I had extensive DM chat with @kurlov about this. Dropping `isEnabled` can be done only if we can prove that CRs created with this attribute will work without any errors after upgrading to the version that doesn't support this flag. If this can be proved reliably, we can drop `isEnabled` already and hardcode the logic that selects Postgres by default.

Reply:
I've tested with dropping the field but I am not 100% sure that I can cover all possible scenario. I think having hidden redundant field is an acceptable trade for have smooth and safe migration. So I lean towards having this field and dropping it in the future

Reply:
Sorry, I don't quite follow the logic. Can you guys help clarify, please?

1. If we **drop** the `isEnabled` field, we need to be **absolutely** sure we can do it **safely**. Correct?
2. If we need to be absolutely sure we can safely drop `isEnabled`, it does not change if we decide to drop it in the future. Correct?
3. If the above is valid, and we simply remove `isEnabled` in the future without making any validations, there's a chance we break some users that have upgraded from 3.73.

By saying

> So I lean towards having this field and dropping it in the future

you mean that you **also** defer the **work of verifying** whether we can **safely** drop `isEnabled` to the future, correct?

Reply:
>2. If we need to be absolutely sure we can safely drop `isEnabled`, it does not change if we decide to drop it in the future. Correct?

I am not sure it's correct. I think there is an important difference between dropping right away and doing it in the next release. Basically, instead of `3.73 using isEnabled -> 3.74 dropping isEnabled` there will be `3.73 using isEnabled -> 3.74 ignoring isEnabled -> 3.75 dropping isEnabled`. This additional state helps to make sure that nothing should break because of missing `isEnabled`. If such approach sounds redundant I can drop it in this PR

Reply:
The thing is: I'm not sure how `dropping isEnabled` would look like. Will the operator be able to modify Central Custom Resource that has `isEnabled` attribute from 3.73 version? If not, and the flag will remain present in the Custom Resource, will the operator freak out when it encounters this attribute?

Reply:
The operator works normally if there is unknown field in the central CRD yaml. If `isEnabled` is introduced manually or inherited in 3.75 then this field will be considered as unknown and should not violate operator flow

Reply:
> central CRD yaml

Do you mean Central CR or Central CRD? The difference is significant.

I believe the operator, when deployed, brings CRDs with it. CRD defines validation schema for CRs.
If we remove `isEnabled` in, say, 3.75, its CRD will not have a definition for `isEnabled`. However, Central CRs that were created at the time of 3.73 version will contain `isEnabled` attribute. Those CRs will now be validated against the new CRD without the attribute.
I think the CR/CRD validation is not done by the operator, rather by Kubernetes. Yet, the operator does its own pass over the CR when loading it. Would both things be happy about CR having an unrecognized attribute?

Reply:
Sorry, I've meant CR. I totally agree that there are uncertainties.
@SimonBaeumer could you please your input?

Reply:
Removing a field from a CRD is considered a breaking API change. If the removal does not have __any__ side-effects we could do it but proving it is hard.
Therefore the only way to remove a field safely from a CRD is by promoting our CRD API version from `v1alpha1` to `v1alpha2` or `v1beta`.

imho the solution, defaulting to enabled and printing a message if a misconfiguration was applied is perfectly fine.

Going forward I would like to reconsider using annotations for temporary API fields (similar to feature flags) instead of adding them to the public API directly which would allow safe removals.

Reply:
> Going forward I would like to reconsider using annotations for temporary API fields (similar to feature flags) instead of adding them to the public API directly which would allow safe removals.

Indeed. We could have ab-used environment variables mechanism on Central CR to enable CentralDB.

Reply:
thanks for clarification, let's keep `isEnabled` and drop it if we will bump the CRD API version.

Reply:
BTW this was the idea from the beginning, to have `isEnabled` with the values `Default` and `Enabled`, so that we can keep the API the same, and just change the meaning of `Default` from `Disabled` to `Enabled`.

Reply:
@kurlov I want to make a blind suggestion without looking at the code about this:

> thanks for clarification, let's keep isEnabled and drop it if we will bump the CRD API version.

Could you please write a code comment for future-us on `isEnabled` that tells exactly what we learned here? Something in the spirit of

```go
// isEnabled is effectively no-op starting from the version 3.74.0. It should be removed when we
// bump API version of ACS custom resources. Removing it before is unsafe and may break compatibility.
// TODO: Remove isEnabled when bumping API version.
isEnabled attribute declared here
```